File: app/src/main/java/sakaitakao/android/permissionsearch/entity/ContextMenuItem.java
package sakaitakao.android.permissionsearch.entity;

import android.view.Menu;
import android.view.MenuItem;
import android.view.MenuItem.OnMenuItemClickListener;

/**
 * コンテキストメニュー項目
 * 
 * @author takao
 * 
 */
public class ContextMenuItem {

	public int groupId;
	public int itemId;
	public int order;
	public int titleRes;
	public int iconRes;
	public OnMenuItemClickListener listener;

	/**
	 * コンストラクタ
	 */
	public ContextMenuItem() {
	}

	public ContextMenuItem(int itemId, int titleRes, int iconRes, OnMenuItemClickListener listener) {
		this.groupId = Menu.NONE;
		this.itemId = itemId;
		this.order = Menu.NONE;
		this.titleRes = titleRes;
		this.iconRes = iconRes;
		this.listener = listener;
	}

	/**
	 * @param menu
	 * @return
	 */
	public MenuItem addToMenu(Menu menu) {
		MenuItem menuItem = menu.add(groupId, itemId, order, titleRes);
		if (iconRes != 0) {
			menuItem.setIcon(iconRes);
		}
		menuItem.setOnMenuItemClickListener(listener);
		return menuItem;
	}
}

File: app/src/main/kotlin/sakaitakao/android/permissionsearch/entity/ContextMenuItem.kt
package sakaitakao.android.permissionsearch.entity

import android.view.Menu
import android.view.MenuItem
import android.view.MenuItem.OnMenuItemClickListener

/**
 * コンテキストメニュー項目

 * @author takao
 */
class ContextMenuItem {

    val groupId: Int
    val itemId: Int
    val order: Int
    val titleRes: Int
    val iconRes: Int
    val listener: OnMenuItemClickListener

    /**
     * コンストラクタ
     */
    constructor(itemId: Int, titleRes: Int, iconRes: Int, listener: OnMenuItemClickListener) {
        this.groupId = Menu.NONE
        this.itemId = itemId
        this.order = Menu.NONE
        this.titleRes = titleRes
        this.iconRes = iconRes
        this.listener = listener
    }

    /**
     * @param menu
     * *
     * @return
     */
    fun addToMenu(menu: Menu): MenuItem {
        val menuItem = menu.add(groupId, itemId, order, titleRes)
        if (iconRes != 0) {
            menuItem.setIcon(iconRes)
        }
        menuItem.setOnMenuItemClickListener(listener)
        return menuItem
    }
}
